### PR TITLE
WT-12672 Extend configuration compilation benchmark to run with multiple threads

### PR DESCRIPTION
**Pull request by ddanderson**

### CODE REVIEW

File: test/csuite/wt11126_compile_config/main.c [hunk 1]
```diff
@@ -316,39 +350,60 @@ int
 main(int argc, char *argv[])
 {
     TEST_OPTS *opts, _opts;
-    uint64_t base_ns, ns, nsecs[N_VARIANTS];
-    int variant, runs;
+    uint64_t base_ns, ns, *nsecs, total_nsecs;
+    uint32_t i, nthreads, variant;
     const char *compiled_config, **compiled_config_array;
+    THREAD_OPTS *thread_opts;
```
Comment (sueloverso) [inline test/csuite/wt11126_compile_config/main.c:356]: Put THREAD_OPTS up below TEST_OPTS.

File: test/csuite/wt11126_compile_config/main.c [hunk 2]
```diff
@@ -316,39 +350,60 @@ int
...
+        thread_opts[i].compiled_config_array = compiled_config_array;
+        thread_opts[i].nsecs = &nsecs[i * N_VARIANTS];
+        pthread_create(&tids[i], NULL, benchmark_thread, &thread_opts[i]);
+    }
+    for (i = 0; i < nthreads; ++i)
+        pthread_join(tids[i], NULL);
 
     printf("number of calls: %d\n", N_CALLS * N_RUNS);
-    base_ns = ns = nsecs[0] / (N_CALLS * N_RUNS);
+    printf("number of threads: %" PRIu32 "\n", nthreads);
```
Comment (sueloverso) [inline test/csuite/wt11126_compile_config/main.c:389]: Should this be above the create/join to give output before the benchmark starts?

**Review by sueloverso — APPROVED** [review]
LGTM with a couple minor comments.

File: test/csuite/wt11126_compile_config/main.c (hunk 1 above)
Comment (ddanderson) [reply test/csuite/wt11126_compile_config/main.c]: Thanks, fixed.

**Pull request merged**